pkg/collector/metadata/agentchecks/agentchecks.go

```diff
@@ -63,7 +63,7 @@ func GetPayload(ctx context.Context) *Payload {
 		agentChecksPayload.AgentChecks = append(agentChecksPayload.AgentChecks, status)
 	}
 
-	configErrors := autodiscovery.GetConfigErrors()
+	configErrors := autodiscoveryimpl.GetConfigErrors()
```

Comment:
Shouldn't this be calling a component method, instead of a package-global function?

Reply:
It is a pkg global function before the migration. It would be painful to pass the component into GetPayload here. It is similar to GetLoaderErrors and other global functions called in this GetPayload function.